python/src/iceberg/expressions/base.py

```diff
@@ -328,3 +329,49 @@ def eval(self, struct: StructProtocol) -> Any:
             Any: The value at the referenced field's position in `struct`
         """
         return self._accessor.get(struct)
+
+
+class UnboundReference:
+    """A reference not yet bound to a field in a schema
+
+    Args:
+        name (str): The name of the field
```

Comment:
it might pay to provide more details on name here for nested fields, or at least point to other documentation on how they are expected to be referred to.

This isn't directly related to this PR, but i could find in the specification what is considered a valid field name other than the fact that they can contain periods "." at least for imports.  This is quite possibly user error though.

Reply:
There aren't spec requirements for field names. In the implementations, we index using `.` to join the names so that you can easily project names like `a.b`.

Reply:
@emkornfield agreed this could use more details. Once the accessor functionality is added I'm planning to circle back here and add an example in the docstring along the lines of:
```py
    Example:
        >>> from iceberg.expressions.base import UnboundReference
        >>> from iceberg.schema import Schema
        >>> from iceberg.types import FloatType, MapType, NestedField, StringType
        >>> schema = Schema(
        ...     NestedField(
        ...         field_id=1,
        ...         name="location",
        ...         field_type=MapType(
        ...             key_id=2,
        ...             key_type=StringType(),
        ...             value_id=3,
        ...             value_type=FloatType(),
        ...             value_is_optional=False,
        ...         ),
        ...         is_optional=False,
        ...     ),
        ...     schema_id=1,
        ... )
        >>> name = "location.lat"
        >>> unbound_reference = UnboundReference(name="location.lat")
        >>> unbound_reference.bind(schema=schema, case_sensitive=False)
        BoundReference(field=1, accessor=Accessor(position=1, inner=Accessor(position=2)))
```

I'll also expand name description to be more descriptive. Thanks!